Firmware/Marlin_main.cpp

```diff
@@ -7672,24 +7672,23 @@ SERIAL_PROTOCOLPGM("\n\n");
     case 125:
     case 601:
     {
-        if (code_seen('X')) {
-          //Check that the X pause position is within printer limits
-          if ((code_value() >= X_MIN_POS) && (code_value() <= X_MAX_POS-1)) {
-            pause_position[X_AXIS] = code_value();
-          } else pause_position[X_AXIS] = X_PAUSE_POS;
+        float temp_pause_position[3] = { X_PAUSE_POS, Y_PAUSE_POS, Z_PAUSE_LIFT };
+        for (uint8_t axis = 0; axis < E_AXIS; axis++) {
+          if (code_seen(axis_codes[axis])) temp_pause_position[axis] = code_value();
         }
-        if (code_seen('Y')) {
-          //Check that the Y pause position is within printer limits
-          if ((code_value() >= Y_MIN_POS) && (code_value() <= Y_MAX_POS-1)) {
-            pause_position[Y_AXIS] = code_value();
-          } else pause_position[Y_AXIS] = Y_PAUSE_POS;
+        //Check that the X pause position is within printer limits
+        if ((temp_pause_position[X_AXIS] >= X_MIN_POS) && (temp_pause_position[X_AXIS] <= X_MAX_POS-1)) {
+            pause_position[X_AXIS] = temp_pause_position[X_AXIS];
+          }
+        //Check that the Y pause position is within printer limits
+        if ((temp_pause_position[Y_AXIS] >= Y_MIN_POS) && (temp_pause_position[Y_AXIS] <= Y_MAX_POS-1)) {
+          pause_position[Y_AXIS] = temp_pause_position[Y_AXIS];
         }
-        if (code_seen('Z')) {
-          //Check that the Z pause lift is within printer limits
-          if (code_value() <=Z_MAX_POS) {
-            pause_position[Z_AXIS] = code_value();
-          } else pause_position[Z_AXIS] = Z_PAUSE_LIFT;
+        //Check that the Z lift position is within printer limits
+        if (temp_pause_position[Z_AXIS] <=Z_MAX_POS) {
+          pause_position[Z_AXIS] = temp_pause_position[Z_AXIS];
         }
+
```

Comment:
@3d-gussner 
Replacing all this, with this:

```cpp
        // Set default values
        pause_position[X_AXIS] = X_PAUSE_POS;
        pause_position[Y_AXIS] = Y_PAUSE_POS;
        pause_position[Z_AXIS] = Z_PAUSE_LIFT;

        for (uint8_t axis = 0; axis < E_AXIS; axis++) {
          if (code_seen(axis_codes[axis])) {
            pause_position[axis] = constrain(code_value(), min_pos[axis], max_pos[axis]);
          }
        }
```

Saves 66 bytes of Flash, and 12 bytes of SRAM

I'm assuming here we can use `min_pos` and `max_pos` arrays.